v2/internal/testcommon/vcr/v3/test_recorder_test.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v3
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/gomega"
+
+	"github.com/Azure/azure-service-operator/v2/internal/config"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon/vcr"
+)
+
+//nolint:paralleltest
+func TestRecorder_WhenRecordingAndRecordingDoesNotExist_MakesRecording(t *testing.T) {
+	// NB: Can't run tests using Setenv() in parallel
+	t.Setenv("AZURE_SUBSCRIPTION_ID", "00000000-0000-0000-0000-000000000000")
```

Comment:
Where are these env variables being read? If it's from `config.Values` someplace (I don't see exactly where...) can we not just pass dummy values into the `config.Values` struct directly? That would allow this test to be run in parallel and avoid this awkward env variable mutation.

Reply:
There's a check for these environment variables being present that's triggered when the recording file doesn't exist. I couldn't see an easy way to suppress, so I took this approach. Can revisit in a future PR.